buildkitd/entrypoint.sh

```diff
@@ -36,6 +36,24 @@ if [ -z "$EARTHLY_CACHE_VERSION" ]; then
     exit 1
 fi
 
+if [ -f "/sys/fs/cgroup/cgroup.controllers" ]; then
+    echo "detected cgroups v2"
+
+    mkdir /sys/fs/cgroup/earthly
+    mkdir /sys/fs/cgroup/buildkit
+    echo $$ > /sys/fs/cgroup/earthly/cgroup.procs
+
+    echo "+pids" > /sys/fs/cgroup/cgroup.subtree_control
+    echo "+cpu" > /sys/fs/cgroup/cgroup.subtree_control
+
+    echo "+pids" > /sys/fs/cgroup/buildkit/cgroup.subtree_control
+    echo "+cpu" > /sys/fs/cgroup/buildkit/cgroup.subtree_control
+
+    test "$(cat /sys/fs/cgroup/cgroup.type)" = "domain"
+    test "$(cat /sys/fs/cgroup/earthly/cgroup.type)" = "domain"
+    test "$(cat /sys/fs/cgroup/buildkit/cgroup.type)" = "domain"
```

Comment:
I actually had such a thing initially and was on the fence since the `--debug` flag would `set -x`. I added them back in to make it more clear to our users if they ever do fail.